src/lib.rs

```diff
@@ -154,6 +157,37 @@ fn encrypt<'p>(
     Ok(PyBytes::new(py, &encrypted))
 }
 
+#[pyfunction]
+fn encrypt_file(
+    infile: &PyString,
+    outfile: &PyString,
+    recipients: Vec<Box<dyn PyrageRecipient>>,
+) -> PyResult<()> {
+    // This turns each `dyn PyrageRecipient` into a `dyn Recipient`, which
+    // is what the underlying `age` API expects.
+    let recipients = recipients.into_iter().map(|pr| pr.as_recipient()).collect();
+
+    let reader = File::open(&*infile.to_string_lossy())?;
+    let writer = File::create(&*outfile.to_string_lossy())?;
```

Comment:
Nit: Let's use the fallible `to_str` API here and surface it as a `ValueError` or similar; lossily rewriting the filenames is likely to be surprising to users.

Reply:
Good catch!